api/src/routes/edit.ts

```diff
@@ -126,6 +126,58 @@ function uploadFile(req, res, next) {
     });
 }
 edit.post("/object/:pid/datastream/:stream", requireToken, datastreamSanitizer, uploadFile);
+edit.post(
+    "/object/:pid/datastream/:stream/license",
+    requireToken,
+    bodyParser.json(),
+    datastreamSanitizer,
+    async function (req, res) {
+        const { pid, stream } = req.params;
+        const { licenseKey } = req.body;
+        try {
+            console.log("Uploading license", licenseKey);
+            const datastream = DatastreamManager.getInstance();
+            await datastream.uploadLicense(pid, stream, licenseKey);
+            res.status(200).send("Upload license success");
+        } catch (error) {
+            console.log("error", error);
+            res.status(500).send(error.message);
+        }
+    }
+);
+
+edit.get("/object/:pid/datastream/:stream/license", requireToken, datastreamSanitizer, async (req, res) => {
```

Comment:
You seem to define this same route twice for some reason.

Reply:
One of them is a "GET"  call that reads the particular license. The other one is a "POST" call that updates the particular license.

Reply:
Are you sure? I see two identical `edit.get("/object/:pid/datastream/:stream/license"` definitions here...

Reply:
(Well, the code isn't identical, but the declarations are... If one is supposed to be a POST, shouldn't it be `edit.post`?

Reply:
Oh. hmm. Let me check.

Reply:
Okay. I removed it.